src/System.Management.Automation/namespaces/RegistryProvider.cs

```diff
@@ -3588,7 +3588,16 @@ private IRegistryWrapper GetRegkeyForPath(string path, bool writeAccess)
                                 }
                             }
 
-                            tempKey = currentKey.OpenSubKey(subKey, writeAccess);
+                            try
+                            {
+                                tempKey = currentKey.OpenSubKey(subKey, writeAccess);
+                            }
+                            catch (Exception e)
+                            {
+                                WriteError(new ErrorRecord(e, e.GetType().FullName, ErrorCategory.InvalidOperation, subKey));
```

Comment:
Another change in behavior here is that this appears to be a loop looking for a matching subkey where you can now see non-terminating errors whereas previously it would be silent until the end?